src/main/java/com/google/research/bleth/simulator/Board.java

```diff
@@ -0,0 +1,58 @@
+package com.google.research.bleth.simulator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/** A container for all the agents, representing their locations, either real or estimated. */
+public class Board {
+    public final int rowNum;
+    public final int colNum;
+    ArrayList<Agent>[][] matrix;
```

Comment:
Prefer to avoid Java's primitive array types - I would recommend Guava Table instead.
https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Table.html

Also see go/ej3e-28, http://doc/1LOhv9tywZmQ5shwhFDMpplpBoUiH142hoqSObCQPXKk/view